ios/FluentUI/Controls/Button.swift

```diff
@@ -120,26 +141,35 @@ open class Button: UIButton {
     open override var contentEdgeInsets: UIEdgeInsets {
         didSet {
             isUsingCustomContentEdgeInsets = true
+
+            if !isAdjustingCustomContentEdgeInsetsForImage && image(for: .normal) != nil {
+                adjustCustomContentEdgeInsetsForImage()
+            }
         }
     }
 
     open override var intrinsicContentSize: CGSize {
-        var size = titleLabel?.systemLayoutSizeFitting(CGSize(width: proposedTitleLabelWidth == 0 ? .greatestFiniteMagnitude : proposedTitleLabelWidth, height: .greatestFiniteMagnitude)) ?? .zero
-        size.width = ceil(size.width + contentEdgeInsets.left + contentEdgeInsets.right)
-        size.height = ceil(max(size.height, style.minTitleLabelHeight) + contentEdgeInsets.top + contentEdgeInsets.bottom)
+        var size = super.intrinsicContentSize
+        let labelHeight = titleLabel?.intrinsicContentSize.height ?? 0.0
+        size.height = ceil(max(labelHeight, style.minTitleLabelHeight)) + contentEdgeInsets.top + contentEdgeInsets.bottom
+
         return size
     }
 
-    /// if value is 0.0, CGFloat.greatestFiniteMagnitude is used to calculate the width of the `titleLabel` in `intrinsicContentSize`
-    private var proposedTitleLabelWidth: CGFloat = 0.0 {
```

Comment:
if you are deleting this, please make sure you aren't regressing the fix commit 965e4cf in large text mode

Reply:
Good catch. I thought this was working but it only worked on large enough screens. On iPhone, the button was not sizing correctly. I updated the sizing logic to fix this issues and also make it work when an icon is set.